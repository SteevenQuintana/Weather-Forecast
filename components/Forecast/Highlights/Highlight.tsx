import Image from 'next/image'

interface HighlightProps {
  title: string
  img: string
  description: string | number
  unit: string
}

const Highlight = ({ title, img, description, unit }: HighlightProps) => (
  <div className='w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]'>
    <h3>{title}</h3>
    <Image
      src={img}
      alt={description.toString()}
      height={150}
      width={150}
      className='m-auto'
    />
    <p>
      {description.toString()} <span>{unit}</span>
    </p>
  </div>
)

export default Highlight
