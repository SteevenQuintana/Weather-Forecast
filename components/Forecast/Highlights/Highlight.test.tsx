import '@testing-library/jest-dom/extend-expect'
import { render, screen } from '@testing-library/react'
import Highlight from './Highlight'

jest.mock('next/image', () => ({
  __esModule: true,
  default: (props: any) => {
    // eslint-disable-next-line jsx-a11y/alt-text, @next/next/no-img-element
    return <img {...props} />
  }
}))

const mockHighlight = {
  title: 'Title',
  img: 'img',
  description: 'description',
  unit: '%'
}

describe('Highlight', () => {
  it('should render successfully', () => {
    render(<Highlight {...mockHighlight} />)

    expect(screen.getByText(mockHighlight.title)).toBeInTheDocument()
    expect(screen.getByAltText(mockHighlight.description)).toBeInTheDocument()
    expect(screen.getByText(mockHighlight.description)).toBeInTheDocument()
    expect(screen.getByText(mockHighlight.unit)).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { container } = render(<Highlight {...mockHighlight} />)
    expect(container).toMatchSnapshot()
  })
})
