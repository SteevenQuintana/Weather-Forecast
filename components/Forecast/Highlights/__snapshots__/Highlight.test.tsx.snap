// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Highlight should match snapshot 1`] = `
<div>
  <div
    class="w-full bg-[color:var(--background-body2)] flex flex-col justify-center text-center px-0 py-2 rounded-xl hover:bg-[color:var(--button-hover)] hover:text-[color:var(--text-main)]"
  >
    <h3>
      Title
    </h3>
    <img
      alt="description"
      class="m-auto"
      height="150"
      src="img"
      width="150"
    />
    <p>
      description
       
      <span>
        %
      </span>
    </p>
  </div>
</div>
`;
